back/src/domains/establishment/adapters/PgDiscussionRepository.ts

```diff
@@ -62,21 +62,34 @@ export class PgDiscussionRepository implements DiscussionRepository {
   }
 
   public async getDiscussions(
-    params: GetDiscussionsParams,
+    { createdSince, lastAnsweredByCandidate, sirets }: GetDiscussionsParams,
+    limit: number,
   ): Promise<DiscussionDto[]> {
-    if (params.sirets.length === 0) return [];
-
-    const pgResults = await pipeWithValue(
+    const queryBuilder = pipeWithValue(
       this.#makeDiscussionQueryBuilder(),
-      (builder) =>
-        params.createdSince
-          ? builder.where("discussions.created_at", ">=", params.createdSince)
-          : builder,
-      (builder) =>
-        params.sirets
-          ? builder.where("discussions.siret", "in", params.sirets)
-          : builder,
-    ).execute();
+      (b) =>
+        createdSince
+          ? b.where("discussions.created_at", ">=", createdSince)
+          : b,
+      (b) =>
+        sirets && sirets?.length > 0
+          ? b.where("discussions.siret", "in", sirets)
+          : b,
+      (b) =>
+        lastAnsweredByCandidate
+          ? b
+              .where("last_exchanges.sender", "=", "potentialBeneficiary")
+              .where((eb) =>
+                eb.between(
+                  "last_exchanges.sent_at",
+                  lastAnsweredByCandidate.from,
+                  lastAnsweredByCandidate.to,
+                ),
+              )
+          : b,
+    ).limit(limit);
+
+    const pgResults = await queryBuilder.execute();
```

Comment:
l'objet queryBuilder est peut-être pas utile. 

```suggestion
    const pgResults = await this.#makeDiscussion....(...)
    .limit(limit)
    .execute()
```